OpenRA.Mods.Common/Widgets/Logic/AssetBrowserLogic.cs

```diff
@@ -397,42 +452,76 @@ bool LoadAsset(IReadOnlyPackage package, string filename)
 						prefix += "|";
 				}
 
-				var video = VideoLoader.GetVideo(Game.ModData.DefaultFileSystem.Open(filename), Game.ModData.VideoLoaders);
-				if (video != null)
+				var fileExtension = Path.GetExtension(filename.ToLowerInvariant());
+				if (allowedSpriteExtensions.Contains(fileExtension))
 				{
-					player = panel.Get<VideoPlayerWidget>("PLAYER");
-					player.Load(prefix + filename);
-					player.DrawOverlay = false;
-					isVideoLoaded = true;
+					currentSprites = world.Map.Rules.Sequences.SpriteCache[prefix + filename];
+					currentFrame = 0;
 
 					if (frameSlider != null)
 					{
-						frameSlider.MaximumValue = (float)player.Video.Frames - 1;
-						frameSlider.Ticks = 0;
+						frameSlider.MaximumValue = (float)currentSprites.Length - 1;
+						frameSlider.Ticks = currentSprites.Length;
 					}
 
-					return true;
-				}
+					currentVoxel = null;
 
-				if (Path.GetExtension(filename.ToLowerInvariant()) == ".vxl")
+					// Just in case we're switching away from a type of asset that forced the music to pause.
+					UnMuteSounds();
```

Comment:
Wouldn't it make more sense to do this when "unloading" an asset?

Reply:
Or rather, why don't we do this at the top while cleaning the old asset? I assume that is because the cases below mute right again, but those calls are there anyway. I'd prefer unmuting above for visibility/generality. What do you think?

Reply:
Moved directly to `ClearLoadedAssets()`.